server/web/test/index.js

```diff
@@ -87,6 +88,39 @@ exports.register = function (server, options, next) {
     }
   });
 
+  // allows owner or admin to publish a test making it visible to the public
+  server.route({
+    method: 'GET',
+    path: '/{testSlug}/{rev}/publish',
+    handler: function (request, reply) {
+      pagesService.getBySlug(request.params.testSlug, request.params.rev)
+      .then(function (values) {
+        const page = values[0];
+        const own = request.session.get('own') || {};
+        const isOwn = own[page.id];
+        const isAdmin = request.session.get('admin');
+
+        if (isOwn || isAdmin) {
```

Comment:
link is only displayed if `page.visible === 'n'` https://github.com/jsperf/jsperf.com/blob/2b9d982b66816fd24559e9af04db57c61400f7f3/server/web/test/index.js#L73 not worried about owner or admins hitting URL directly